libc/src/stdio/printf_core/writer.h

```diff
@@ -23,15 +23,19 @@ namespace printf_core {
 
 struct WriteBuffer {
   using StreamWriter = int (*)(cpp::string_view, void *);
+  using ResizeOverflowWriter = int (*)(WriteBuffer *wb,
+                                       cpp::string_view new_str);
   char *buff;
-  const size_t buff_len;
+  size_t buff_len;
   size_t buff_cur = 0;
 
   // The stream writer will be called when the buffer is full. It will be passed
   // string_views to write to the stream.
   StreamWriter stream_writer;
   void *output_target;
 
+  ResizeOverflowWriter resize_writer = nullptr;
```

Comment:
you don't need to add this, just make `output_target` a pointer to this `WriteBuffer` and then reinterpret cast it back inside of the `StreamWriter` function